packages/studio-ui/src/components/AddPageButton/AddPageContextProvider.tsx

```diff
@@ -16,9 +16,22 @@ export default function AddPageContextProvider(props: PropsWithChildren) {
     () => ({
       state,
       actions: {
-        setIsStatic: (isStatic: boolean) => setState({ ...state, isStatic }),
+        setIsStatic: (isStatic: boolean) =>
+          setState((state) => {
+            return { ...state, isStatic };
+          }),
         setStreamScope: (streamScope: StreamScope) =>
-          setState({ ...state, streamScope }),
+          setState((state) => {
+            return { ...state, streamScope };
+          }),
+        setPageName: (pageName: string) =>
+          setState((state) => {
+            return { ...state, pageName };
+          }),
+        setGetPathVal: (getPathVal?: GetPathVal) =>
+          setState((state) => {
+            return { ...state, getPathVal };
+          }),
```

Comment:
If all these are doing is setting values in the state object, I think it'd be simpler to expose setState instead of have 4 different actions

Reply:
oh, that's so true and makes a lot of sense.  We changed the setStates to function setStates to make sure that old values don't get overwritten if react calls them in batches.  Even through we have all state updates in once function, it is ok to keep the functional setState for safety?

Reply:
yeah that's fine!